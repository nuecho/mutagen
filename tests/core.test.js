test("[mutagen] should print usage with banner", () => {
  const {code, output} = mutagen();

  expect(output).toMatchSnapshot("usage with banner");
  expect(code).toBe(0);
});

test("[mutagen --help] should print usage", () => {
  const {code, output} = mutagen(`--help`);

  expect(output).toMatchSnapshot("usage");
  expect(code).toBe(0);
});
