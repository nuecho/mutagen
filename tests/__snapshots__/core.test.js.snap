// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[mutagen --help] should print usage: usage 1`] = `
Object {
  "stderr": "",
  "stdout": "Usage: mutagen [OPTIONS] <command>

      --metrics=<metricsFile> Performance metrics output file.
  -?, -h, --help              Show this help message.
  -d, --debug                 Set log level to debug.
  -e, --env=<environmentName> Environment name used for the execution.
  -i, --info                  Set log level to info.
  -p, --read-password-from-stdin
                              Read password from standard input.
  -s, --stacktrace            Print out the stacktrace for all exceptions.
  -v, --version               Show version info.

Commands:

  agent         Genesys Agents tool
  audio         GAX ARM tool
  config        Genesys Config Server tool
  set-password  Save the password for your env encrypted in a not so secure way
  services      Discover services

See 'mutagen <command> --help' to read about a specific command.
",
}
`;

exports[`[mutagen] should print usage with banner: usage with banner 1`] = `
Object {
  "stderr": "",
  "stdout": "
_   .-')                .-') _      ('-.                   ('-.       .-') _
( '.( OO )_             (  OO) )    ( OO ).-.             _(  OO)     ( OO ) )
 ,--.   ,--.),--. ,--.  /     '._   / . --. /  ,----.    (,------.,--./ ,--,'
 |   \`.'   | |  | |  |  |'--...__)  | \\\\-.  \\\\  '  .-./-')  |  .---'|   \\\\ |  |\\\\
 |         | |  | | .-')'--.  .--'.-'-'  |  | |  |_( O- ) |  |    |    \\\\|  | )
 |  |'.'|  | |  |_|( OO )  |  |    \\\\| |_.'  | |  | .--, \\\\(|  '--. |  .     |/
 |  |   |  | |  | | \`-' /  |  |     |  .-.  |(|  | '. (_/ |  .--' |  |\\\\    |
 |  |   |  |('  '-'(_.-'   |  |     |  | |  | |  '--'  |  |  \`---.|  | \\\\   |
 \`--'   \`--'  \`-----'      \`--'     \`--' \`--'  \`------'   \`------'\`--'  \`--'
 Your Genesys Toolbox. (https://sites.google.com/m.nuecho.com/hub/mutagen)

Usage: mutagen [OPTIONS] <command>

      --metrics=<metricsFile> Performance metrics output file.
  -?, -h, --help              Show this help message.
  -d, --debug                 Set log level to debug.
  -e, --env=<environmentName> Environment name used for the execution.
  -i, --info                  Set log level to info.
  -p, --read-password-from-stdin
                              Read password from standard input.
  -s, --stacktrace            Print out the stacktrace for all exceptions.
  -v, --version               Show version info.

Commands:

  agent         Genesys Agents tool
  audio         GAX ARM tool
  config        Genesys Config Server tool
  set-password  Save the password for your env encrypted in a not so secure way
  services      Discover services

See 'mutagen <command> --help' to read about a specific command.

Please specify a command.
",
}
`;
